### PR TITLE
Christmas cleanup and modularisations

### PR DESCRIPTION
**Pull request by guillep**
- moving Dependent out of object
- removing dead classes
- moving date related classes to the chronology package

### CODE REVIEW

**Comment by Ducasse** [comment]
Did not bootstrap.

**Comment by Ducasse** [comment]
```
Looking for module  ... loaded...computing accessor depths17:00:38		[+] Creating image on which bootstrapping will take place

warning, variable nsMethodCache doesn't exist or has already been removed
...done17:00:41		Initializing bootstrap environment.
17:00:41		Creating initial Objects needed by the VM
17:00:41		Flushing new space.
17:00:41				Begin: Initializing initial objects
17:00:41				Initializing objects.
17:00:41				initializing hash table sizes
17:00:41				preparing class builder
17:00:42				class loader now creates class pools
17:00:42				Initializing StartUp list
17:00:42				Initializing Character Table
17:00:42				Initializing String Ascii and CaseInsensitive Table
NewUndeclaredWarning: ProtoObject>>noMethod (DateAndTime is Undeclared)
Instance of EPASTInterpreter did not understand #visitLiteralVariableNode:
EPASTInterpreter(Object)>>doesNotUnderstand: #visitLiteralVariableNode:
UndeclaredVariable(LiteralVariable)>>acceptVisitor:node:
RBVariableNode>>acceptVisitor:
EPASTInterpreter(ASTInterpreter)>>interpret:
EPASTInterpreter>>visitMessageNode:
RBMessageNode>>acceptVisitor:
EPASTInterpreter(ASTInterpreter)>>interpret:
EPASTInterpreter>>evaluateAST:
EPASTEvaluator>>evaluateAST:
EPASTEvaluator>>evaluateCode:withTemps:
EPASTEvaluator>>evaluateCode:
17:00:43				Initializing SmallInteger constants
17:00:43				Initializing Date and Time constants
[
```

File: src/System-SessionManager/WorkingSession.class.st [hunk 1]
```diff
@@ -64,7 +58,6 @@ WorkingSession >> executeDeferredStartupActions: resuming [
 WorkingSession >> initialize [
 	super initialize.
 	deferredStartupActions := OrderedCollection new.
-	creationTime := DateAndTime now.
```
Comment (Ducasse) [inline src/System-SessionManager/WorkingSession.class.st:67]: I wondered if this was used. 
In the redefinition later will we reintroduce it?

Comment (Ducasse) [reply src/System-SessionManager/WorkingSession.class.st:67]: Probably for sorting them.

**Comment by Ducasse** [comment]
The class side initialize are surprising to me. Did you check the complete system?

**Comment by Ducasse** [comment]
I was thinking about the `Delay forDuration: ` and I will open an issue. Because we have totally stupid code

```
indicateModalChild
	"Flash the button border."

	|fs c w d|
	fs := self fillStyle.
	c := self color alphaMixed: 0.5 with: Color black.
	w := self world.
	d := 0.
	self assureExtension.
	2 timesRepeat: [
		(Delay forDuration: d milliSeconds) wait.
		d := 200.
		extension fillStyle: c.
		self privateColor: c.
		self invalidRect: self bounds.
		w ifNotNil: [w displayWorldSafely].
		(Delay forDuration: d milliSeconds) wait.
		self fillStyle: fs.
		w ifNotNil: [w displayWorldSafely].
		self invalidRect: self bounds]
```

File: src/System-SessionManager/WorkingSession.class.st (hunk 1 above)
Comment (guillep) [reply src/System-SessionManager/WorkingSession.class.st:67]: Nope, not even...

**Comment by guillep** [comment]
> The class side initialize are surprising to me. Did you check the complete system?

Nope, but I'm positive I caught all of them:

It used to work because `Object class >> initialize` was defined (and resetting the dependents array!).
Since we removed that method, now lookup goes up and clears the class, so it seems that I caught all the extra initializes because the build works :)

**Comment by guillep** [comment]
Broken tests are unrelated

**Comment by guillep** [comment]
Ok, this has introduced a problem when starting up that is not shown in the tests ... I'll revert and rework.

**Pull request merged**